src/rgw/rgw_basic_types.h

```diff
@@ -649,5 +241,54 @@ inline std::ostream& operator<<(std::ostream& out, const rgw_user &u) {
   return out << s;
 }
 
+#include "rgw_obj_manifest.h"
```

Comment:
why not include in the beginning of the file?

Reply:
there's a reason, but it's not good, will revisit